docs/tutorial/frontend-tut.md

```diff
@@ -1,93 +1,158 @@
 # Building a Frontend for Calyx
 
-In this tutorial, we're going to build a compiler for a small language called MrXL.
+> This tutorial assumes that you have already worked through the [Calyx tutorial][calyx-tut]. You won't get very much out of it by itself!
 
-## MrXL Overview
-MrXL provides constructs to create arrays, and perform `map` and `reduce` operations on those arrays. Here's an example of a dot product implementation in MrXL:
+In the [Calyx tutorial][calyx-tut] you wrote Calyx by hand.
+That is (probably) a good way to build character, but it's no way to live.
+In practice, you want a *frontend* that *compiles* to Calyx.
+
+This allows you to:
+1. Support domain-specific commands that Calyx does not.
+2. Automatically generate some of the kludge that Calyx requires.
```

Comment:
Maybe a more direct way to get this point across: Calyx is a compiler IL (not a human-facing language). So let's build a compiler using it.

That is, the point is less about alleviating the pain of writing Calyx by hand… it's about getting to what people showed up to this tutorial to do in the first place, which is to implement a DSL-to-hardware compiler.